seismicpro/refractor_velocity/metrics.py

```diff
@@ -0,0 +1,494 @@
+        _ = args, kwargs
+        raise NotImplementedError
+
+    def __call__(self, *args, **kwargs):
+        """Aggregate the metric. If not overriden, takes mean value of `calc`."""
+        return np.mean(self.calc(*args, **kwargs))
+
+    def plot_gather(
+        self, coords, ax, index, sort_by=None, mask=True, top_header=True, **kwargs):
+        """Base view for gather plotting. Plot the gather by its index in bounded survey and its first breaks.
+        By default also recalculates metric in order to display `top_header` with metric values above gather traces
+        and mark trases with metric greater than threshold. Threshold is either aquired from `kwargs` if given,
+        or metric's colorbar margin if defined, or simply by mean value.
+
+        Parameters
+        ----------
+        sort_by : str or iterable of str, optional.
+            Headers names to sort the gather by.
+        mask : bool, optional, defaults to True.
+            Whether to plot mask defined by metric values on top of the gather plot.
+        top_header : bool, optional, defaults to True.
+            Whether to show a header with metric values above the gather plot.
+        kwargs : misc, optional.
+            Additional keyword arguments to `gather.plot`
+        """
+        _ = coords
+        gather = self.survey.get_gather(index)
+        if sort_by is not None:
+            gather = gather.sort(by=sort_by)
+        event_headers = kwargs.pop("event_headers", {"headers": self.first_breaks_col})
+        if top_header or mask:
+            refractor_velocity = self.field(gather.coords)
+            metric_values = self.calc(gather=gather, refractor_velocity=refractor_velocity,
+                                      first_breaks_col=self.first_breaks_col, correct_uphole=self.correct_uphole)
+            if mask:
+                mask_kwargs = kwargs.get("masks", {})
+                invert_mask = -1 if self.is_lower_better is False else 1
+                mask_threshold = get_first_defined(mask_kwargs.get("threshold", None),
+                                                   self.vmax if invert_mask == 1 else self.vmin,
+                                                   metric_values.mean())
+                mask_kwargs.update({"masks": metric_values * invert_mask,
+                                    "threshold": mask_threshold * invert_mask})
+                kwargs["masks"] = mask_kwargs
+            if top_header:
+                kwargs["top_header"] = metric_values
+        gather.plot(event_headers=event_headers, ax=ax, **kwargs)
+
+    def plot_refractor_velocity(self, coords, ax, index, **kwargs):
+        """Plot the refractor velocity curve."""
+        refractor_velocity = self.field(coords)
+        gather = self.survey.get_gather(index)
+        refractor_velocity.times = gather[self.first_breaks_col]
+        if self.correct_uphole:
+            refractor_velocity.times += gather["SourceUpholeTime"]
```

Comment:
This may update gather headers, I'd force a copy to make it safe
```suggestion
            refractor_velocity.times = refractor_velocity.times + gather["SourceUpholeTime"]
```